### PR TITLE
Refactor Paths and add property filtering

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes
- refactor paths to use hook with kea logic
- fixes a bug where paths weren't always loading (because the sankey and d3 librariers weren't always loading on time)
- add property filtering on paths
- address #849 

## Checklist
- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

**Comment by timgl** [comment]
This is close, but what I actually want is to be able to filter the _first_ item. For example, in this view I want to only show people that start at posthog.com, but I do want to see them end up at app.posthog.com. Currently if I filter on posthog.com it'll filter app.posthog.com out
image

**Comment by EDsCODE** [comment]
Ah, yeah still working out the best way to do that. If this refactor/adjustment is well contained and good to go can we merge this first? 

Atm we can do things like this which will help us look through different types of data
Screen Shot 2020-05-27 at 2 46 53 PM

**Update: Did some deeper QA and finding a few errors with specific filters so no need to QA again yet**

File: posthog/api/paths.py [hunk 1]
```diff
@@ -22,34 +24,40 @@ def _determine_path_type(self, request):
...
     def list(self, request):
         team = request.user.team_set.get()
         resp = []
         date_query = request_to_date_query(request.GET)
-        event, path_type = self._determine_path_type(request)
+        event, path_type, event_filter = self._determine_path_type(request)
+        properties = request.GET.get('properties')
 
-        sessions = Event.objects.filter(
+        sessions = Event.objects.add_person_id(team.pk).filter(
```
Comment (EDsCODE) [inline posthog/api/paths.py:55]: was slightly concerned about this but it didn't have any meaningful effect on query time from eyeballing so i didn't condition it out

Comment (timgl) [reply posthog/api/paths.py:55]: Django ORM only actually adds the person ID when required (eg when filtering on person property)

**Review by timgl — APPROVED** [review]

**Pull request merged**